src/rendering/software/tests.rs

```diff
@@ -200,26 +200,46 @@ fn horizontal() {
         &[Some(10.0), None, Some(20.0), Some(55.0)],
     );
 
-    check_dims(&layout.state(&timer), [1500, 40], 0x3a70ef32, "horizontal");
+    check_dims(&layout.state(&timer), [1500, 40], "YmJicmJSUmM=", "horizontal");
 }
 
-fn check(state: &LayoutState, expected_checksum: u32, name: &str) {
-    check_dims(state, [300, 500], expected_checksum, name);
+fn get_comparison_tolerance() -> u32 {
+    // Without MMX the floating point calculations don't follow IEEE 754, so the tests require a
+    // tolerance that is greater than 0.
+    // FIXME: We use SSE as an approximation for the cfg because MMX isn't supported by Rust yet.
+    if cfg!(all(
+        target_arch = "x86",	
+        not(target_feature = "sse"),	
+    )) {
+        4
+    } else {
+        0
+    }
+}
+
+fn check(state: &LayoutState, expected_hash_data: &str, name: &str) {
+    check_dims(state, [300, 500], expected_hash_data, name);
 }
 
-fn check_dims(state: &LayoutState, dims: [usize; 2], expected_checksum: u32, name: &str) {
+fn check_dims(state: &LayoutState, dims: [usize; 2], expected_hash_data: &str, name: &str) {
     let image = render(state, dims);
-    let calculated_checksum = checksum_ieee(&image);
+    let hasher = HasherConfig::with_bytes_type::<[u8; 8]>().to_hasher();
+
+    let calculated_hash = hasher.hash_image(&image);	
+    let calculated_hash_data = calculated_hash.to_base64();
+    let expected_hash = ImageHash::<[u8; 8]>::from_base64(expected_hash_data).unwrap();	
+    let distance = calculated_hash.dist(&expected_hash);
 
     fs::create_dir_all("target/renders").ok();
 
-    let path = format!("target/renders/{}_{:08x}.png", name, calculated_checksum);
-    image.save(&path).ok();
+    let expected_path = format!("target/renders/{}_expected.png", name);
+
+    if distance > get_comparison_tolerance() {
+        let actual_path = format!("target/renders/{}_actual.png", name);
```

Comment:
I'm not sure if I'm too much of a fan of this. The old one had the advantage that I could switch branches and it still would find the expected image for the specific branch, so if visual changes would happen, it's able to create a proper diff.

Reply:
What we could do is throw the hash bytes into AES to get highly scrambled bytes out again and revert the AES to get back the image hash for the expected AES hash.

Reply:
what stops us from using hex instead of base64?

Reply:
For now, I reverted to the old behavior of using the hash in the filename, but with the `/` all replaced with `$` so that there's no issue with certain filenames. We can use AES if we think that the hashes being too similar is a problem (but I'm also not sure how to use the `aes` crate).